### ISSUE TITLE
Maps played in playlists don't update their 'Last Played' status

### ISSUE THREAD

**Issue by vis256** [ppy/osu#28013]
### Type

Game behaviour

### Bug description

When playing a map you already own in a playlist I think it should go to the top of the list if sorted by 'Last Played' status.

### Version

2024.421-lazer

### Logs

[compressed-logs.zip](https://github.com/ppy/osu/files/15139606/compressed-logs.zip)

**Comment by bdach** [comment]
I can't reproduce this. Works just fine here.

@vis256 can you describe the steps you took to discover this? Maybe this is due to lack of re-sort on the carousel or something?

**Comment by vis256** [comment]
https://github.com/ppy/osu/assets/67900555/92cd53b4-9277-44c6-a9ab-1bc3fc252f1a

Excuse my bad gameplay, you can skip 00:15 - 02:50.
Also I think the song's position is now accurate after restarting the game?

**Comment by bdach** [comment]
> Also I think the song's position is now accurate after restarting the game?

Yeah okay this is going to be something or other about the sort not applying again in the actual game even though the date updated fine in the database.
